packages/shared/pkg/feature-flags/client.go

```diff
@@ -51,28 +56,36 @@ func NewClient() (*Client, error) {
 	return &Client{ld: ldClient}, nil
 }
 
-func (c *Client) BoolFlag(flag BoolFlag, contextKey string) (bool, error) {
+func (c *Client) BoolFlag(ctx context.Context, flag BoolFlag, contextKey string) (bool, error) {
 	if c.ld == nil {
 		return flag.fallback, fmt.Errorf("LaunchDarkly client is not initialized")
 	}
 
 	flagCtx := ldcontext.NewBuilder(flag.name).SetString("contextKey", contextKey).Build()
-	enabled, err := c.ld.BoolVariation(flag.name, flagCtx, flag.fallback)
+	if embeddedCtx, ok := getContext(ctx); ok {
+		flagCtx = ldcontext.NewMulti(embeddedCtx, flagCtx)
+	}
```

Comment:
This ensures backwards compatibility in case we were using the contextKey somewhere; `ldcontext.NewBuilder(...)` creates a context with `kind=user` and assigns the string as its key. The code here keeps that alive, but adds to it whatever context we've attached to the `ctx` parameter.